File: 1859-sorting-the-sentence/1859-sorting-the-sentence.py
class Solution:
    def sortSentence(self, s: str) -> str:
        s_split = s.split()  # TC: O(n)
        res = [0]*len(s_split) # SC: O(n)
        for word in s_split:
            res[int(word[-1])-1] = word[:-1]
        return " ".join(res)
    
# Time Complexity : O(n)
# Space Complexity : O(n)

File: 1859-sorting-the-sentence/1859-sorting-the-sentence.java
class Solution {
    public String sortSentence(String s) {
	    String[] s_split = s.split(" ");
        String[] res = new String[s_split.length];
        for(String word : s_split){
            res[(int)word.charAt(word.length()-1)-48-1] = word.substring(0, word.length()-1);           
        }
        return String.join(" ", res);
    }
}

// Time Complexity: O(n)
// Space Complexitu: O(n)